filetablemodel.cpp

```diff
@@ -106,10 +107,23 @@ QVariant FileTableModel::data(const QModelIndex &index,
                               int role) const
 {
     // We validate arguments.
-    if (!index.isValid() || role != Qt::DisplayRole ||
-            index.column() >= columnCount() || index.row() >= rowCount())
+    if (!index.isValid() || index.column() >= columnCount() || index.row() >= rowCount())
+        return QVariant();
+    switch (role) {
+    case Qt::DisplayRole:
+        return m_data.at(index.column()).at(index.row());
+        break;
+    case Qt::CheckStateRole:
+        if ((index.column() == 0) && (index.row() < m_checkedStates.count())) {
+            return m_checkedStates[index.row()];
+        }
+        else {
```

Comment:
This is hard for me to read, since you're not following the style. I suppose you haven't set your Qt Creator settings in regards to the opening braces. Do you want to continue using this style or switch to the one I proposed. I'm fine with whichever you chose, but we need to settle on one.